### ISSUE TITLE
Error while running BasicVSR demo 

### ISSUE THREAD

**Issue by Kadakol** [open-mmlab/mmediting#323]
Hi,

Thank you so much for your work on BasicVSR!

Installation was done as per the [guide](https://github.com/open-mmlab/mmediting/blob/master/docs/install.md)
I was trying to run the demo as per this [document](https://mmediting.readthedocs.io/en/latest/demo.html)

Following is the command I used:

`python demo/restoration_demo.py configs/restorers/basicvsr/basicvsr_reds4.py models/basicvsr_reds4_20120409-0e599677.pth <PATH_TO_DATA>/calendar/ ./basic_vsr/vid4/calendar`

Unfortunately, I get the following error:

```
2021-05-24 15:11:05,378 - mmedit - INFO - Use load_from_http loader
Use load_from_local loader
The model and loaded state dict do not match exactly

missing keys in source state_dict: step_counter

Traceback (most recent call last):
  File "demo/restoration_demo.py", line 38, in <module>
    main()
  File "demo/restoration_demo.py", line 29, in main
    output = restoration_inference(model, args.img_path)
  File "<PATH>/mmediting/mmedit/apis/restoration_inference.py", line 35, in restoration_inference
    data = test_pipeline(data)
  File "<PATH>/mmediting/mmedit/datasets/pipelines/compose.py", line 41, in __call__
    data = t(data)
  File "<PATH>/mmediting/mmedit/datasets/pipelines/augmentation.py", line 946, in __call__
    clip_name = results['key']
KeyError: 'key'
Segmentation fault (core dumped)
```

Please suggest how to fix this issue. Also do let me know if any information is required from my side.

Thanks!

**Comment by ckkelvinchan** [comment]
You should use `restoration_video_demo.py` instead of `restoration_demo.py`.

**Comment by innerlee** [comment]
Add some friendly warnings in `restoration_demo.py` if multiple frames are detected? @ckkelvinchan

**Comment by ckkelvinchan** [comment]
> Add some friendly warnings in `restoration_demo.py` if multiple frames are detected? @ckkelvinchan

That is a good idea.

**Comment by Kadakol** [comment]
Hi @ckkelvinchan ,

Thanks for such a quick response! That was very silly of me. I misread the instructions! :(

I tried the following command:
`python demo/restoration_video_demo.py configs/restorers/basicvsr/basicvsr_reds4.py models/basicvsr_reds4_20120409-0e599677.pth <PATH_TO_DATA>/calendar ./basic_vsr/vid4/calendar/`

I am now getting the following error:

```
2021-05-24 15:41:12,233 - mmedit - INFO - Use load_from_http loader
Use load_from_local loader
The model and loaded state dict do not match exactly

missing keys in source state_dict: step_counter

Traceback (most recent call last):
  File "demo/restoration_video_demo.py", line 43, in <module>
    main()
  File "demo/restoration_video_demo.py", line 33, in main
    args.window_size)
  File "<PATH>/mmediting/mmedit/apis/restoration_video_inference.py", line 61, in restoration_video_inference
    data = test_pipeline(data)
  File "<PATH>/mmediting/mmedit/datasets/pipelines/compose.py", line 41, in __call__
    data = t(data)
  File "<PATH>/mmediting/mmedit/datasets/pipelines/loading.py", line 143, in __call__
    img_bytes = self.file_client.get(filepath)
  File "/home/akshay/anaconda3/envs/basicvsr/lib/python3.7/site-packages/mmcv/fileio/file_client.py", line 306, in get
    return self.client.get(filepath)
  File "/home/akshay/anaconda3/envs/basicvsr/lib/python3.7/site-packages/mmcv/fileio/file_client.py", line 184, in get
    with open(filepath, 'rb') as f:
FileNotFoundError: [Errno 2] No such file or directory: '<PATH_TO_DATA>/calendar/00000000.png'
Segmentation fault (core dumped)
```

It makes sense, there are no files of the format 00000000.png in the calendar dataset. The files are of the format 00xx.png. Could you please suggest how to make this work?

**Comment by innerlee** [comment]
You can change the template

**Comment by ckkelvinchan** [comment]
> You can change the template

For now, `GenerateSegmentIndices` hard-coded to read `{v:08d}`. 

@Kadakol You may need to rename the files for now, and I will fix this as soon as possible.

**Comment by Kadakol** [comment]
Thank you @ckkelvinchan. I tried to modify in a few places but it did not seem to have any effect. Let me try with the renamed files.

**Comment by innerlee** [comment]
Enter a interactive python shell, and `import mmedit; mmedit.__file__` to see if you modified the right place.

**Comment by ckkelvinchan** [comment]
@Kadakol Hi we have just added support for that. In your case, you can use

`python demo/restoration_video_demo.py configs/restorers/basicvsr/basicvsr_reds4.py models/basicvsr_reds4_20120409-0e599677.pth <PATH_TO_DATA>/calendar ./basic_vsr/vid4/calendar/ --filename_tmpl={:04d}.png`

Please let us know if there are any other problems.

**Comment by Kadakol** [comment]
I tried modifying the file names to have 8 digits as well as the latest committed code. They both work. Thank you so much for your response!

However, I came across an observation. Following is the log I get at the end of the run. It doesn't affect the outputs, so it is not an issue; it is just an observation.

```
loading - filepath = <PATH_TO_DATA>/calendar_0_beginning/0039.png
loading - filepath = <PATH_TO_DATA>/calendar_0_beginning/0040.png
Segmentation fault (core dumped)
```

My issue is resolved for now. Thanks!
